nifi-nar-bundles/nifi-framework-bundle/nifi-framework/nifi-web/nifi-web-ui/src/main/webapp/js/nf/canvas/nf-connection.js

```diff
@@ -1467,6 +1478,33 @@
                         }
                     });
 
+                    // determine whether or not to show the penalized icon
+                    connectionLabelContainer.select('text.penalized-icon')
+                        .classed('hidden', function () {
+                            if (d.permissions.canRead) {
+                                return d.component.flowFileAvailability !== 'HEAD_OF_QUEUE_PENALIZED';
+                            } else {
+                                return true;
+                            }
+                        })
+                        .attr('x', function() {
+                            var offset = 0;
+                            if(!connectionLabelContainer.select('text.expiration-icon').classed('hidden')) {
+                                offset += 16;
+                            }
+                            if(!connectionLabelContainer.select('text.load-balance-icon').classed('hidden')) {
+                                offset += 16;
+                            }
+                            return 208 - offset;
+                        })
+                        .select('title').text(function () {
+                        if (d.permissions.canRead) {
```

Comment:
With the `flowFileAvailability` as part of the status, we do not need to check permissions. Statuses are surfaced regardless of permissions to ensure the user is aware of the state of the data flow even if they don't know what the Processors are doing. When we move this to `updateConnectionStatus` we can remove these `canRead` conditions.